conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -71,4 +94,57 @@ public static <T> LinkedHashSet<T> newLinkedHashSet(Iterable<? extends T> iterab
         }
         return set;
     }
+
+    public static <T> List<T> newList() {
+        return new ArrayList<>();
+    }
+
+    public static <T> List<T> newList(Iterable<? extends T> iterable) {
```

Comment:
Deprecated old methods in favor of more generic factory methods that allow us to change backing list implementation